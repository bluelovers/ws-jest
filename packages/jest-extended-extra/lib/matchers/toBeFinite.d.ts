/// <reference types="jest" />
declare const matcherName: "toBeFinite";
declare global {
    namespace jest {
        interface Matchers<R> {
            [matcherName](): R;
        }
        interface Expect {
            [matcherName](): void;
        }
    }
}
export declare function toBeFinite(this: jest.MatcherContext, received: number): {
    pass: boolean;
    message: () => string;
    actual: number;
    expected: "finite";
    name: "toBeFinite";
};
declare const _default: {
    toBeFinite: typeof toBeFinite;
};
export default _default;
