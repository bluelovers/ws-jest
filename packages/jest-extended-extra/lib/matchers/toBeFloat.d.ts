/// <reference types="jest" />
declare const matcherName: "toBeFloat";
declare global {
    namespace jest {
        interface Matchers<R> {
            [matcherName](): R;
        }
    }
}
export declare const toBeFloat: (this: jest.MatcherContext, received: number) => {
    pass: any;
    message: () => string;
};
declare const _default: {
    toBeFloat: (this: jest.MatcherContext, received: number) => {
        pass: any;
        message: () => string;
    };
};
export default _default;
