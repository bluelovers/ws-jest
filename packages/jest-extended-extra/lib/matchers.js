"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.matchers = void 0;
const toBeFinite_1 = __importDefault(require("./matchers/toBeFinite"));
const toBeFloat_1 = __importDefault(require("./matchers/toBeFloat"));
const toBeInfinity_1 = __importDefault(require("./matchers/toBeInfinity"));
const toBeInteger_1 = __importDefault(require("./matchers/toBeInteger"));
const toBeNegative_1 = __importDefault(require("./matchers/toBeNegative"));
const toBePositive_1 = __importDefault(require("./matchers/toBePositive"));
const toBeZero_1 = __importDefault(require("./matchers/toBeZero"));
exports.matchers = {
    ...toBeFinite_1.default,
    ...toBeFloat_1.default,
    ...toBeInfinity_1.default,
    ...toBeInteger_1.default,
    ...toBeNegative_1.default,
    ...toBePositive_1.default,
    ...toBeZero_1.default
};
exports.default = exports.matchers;
//# sourceMappingURL=matchers.js.map