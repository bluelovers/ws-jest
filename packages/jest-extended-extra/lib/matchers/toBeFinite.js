"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.toBeFinite = void 0;
const jest_util_1 = require("@lazy-assert/jest-util");
const matcherName = 'toBeFinite';
const type = 'finite';
function toBeFinite(received) {
    const pass = isFinite(received);
    return {
        pass,
        message: (0, jest_util_1.autoMessage)(pass, received, matcherName, type),
    };
}
exports.toBeFinite = toBeFinite;
exports.default = {
    toBeFinite,
};
//# sourceMappingURL=toBeFinite.js.map