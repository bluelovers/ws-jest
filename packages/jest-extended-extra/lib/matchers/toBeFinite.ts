import { autoMessage } from '@lazy-assert/jest-util';

const matcherName = 'toBeFinite' as const;
const type = 'finite' as const

declare global
{

	namespace jest
	{

		interface Matchers<R>
		{
			[matcherName](): R;
		}

		interface Expect
		{
			[matcherName](): void;
		}

	}

}

export function toBeFinite(this: jest.MatcherContext, received: number)
{
	const pass = isFinite(received);

	return {
		pass,
		message: autoMessage(pass, received, matcherName, type),
	};
}

export default {
	toBeFinite,
};
