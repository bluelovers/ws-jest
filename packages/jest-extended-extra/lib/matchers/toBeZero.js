"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.toBeZero = void 0;
const msg_1 = require("../util/msg");
const matcherName = 'toBeZero';
const type = 'zero';
function toBeZero(received) {
    const pass = received === 0 || received === -0;
    return {
        pass,
        message: msg_1.autoMessage(pass, received, matcherName, type),
    };
}
exports.toBeZero = toBeZero;
exports.default = {
    toBeZero,
};
//# sourceMappingURL=toBeZero.js.map