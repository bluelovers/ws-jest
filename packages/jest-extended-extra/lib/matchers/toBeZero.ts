import { isZero } from '@lazy-assert/check-basic';
import { autoMessage } from '@lazy-assert/jest-util';

const matcherName = 'toBeZero' as const;
const type = 'zero' as const

declare global
{

	namespace jest
	{

		interface Matchers<R>
		{
			[matcherName](): R;
		}

		interface Expect
		{
			[matcherName](): void;
		}

	}

}

export function toBeZero(this: jest.MatcherContext, received: number)
{
	const pass = isZero(received);

	return {
		pass,
		message: autoMessage(pass, received, matcherName, type),
		actual: received,
		expected: type,
		name: matcherName,
	};
}

export default {
	toBeZero,
};
