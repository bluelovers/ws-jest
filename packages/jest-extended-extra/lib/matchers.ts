

import toBeFloat from './matchers/toBeFloat'
import toBeInteger from './matchers/toBeInteger'
import toBeZero from './matchers/toBeZero'

export const matchers = Object.assign(toBeFloat, toBeInteger, toBeZero)
export default matchers

