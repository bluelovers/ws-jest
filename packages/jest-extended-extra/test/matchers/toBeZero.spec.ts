import { toBeZero } from '../../lib/matchers/toBeZero';

expect.extend({
	toBeZero,
})

test(`0`, () =>
{
	expect(0).toBeZero();
});

test(`-0`, () =>
{
	expect(-0).toBeZero();
});

test(`throw`, () =>
{
	expect(() => expect(-0).not.toBeZero()).toThrowErrorMatchingSnapshot();

});

test(`''`, () =>
{
	expect('').not.toBeZero();
});

test(`'0'`, () =>
{
	expect('0').not.toBeZero();
});

test(`'-0'`, () =>
{
	expect('-0').not.toBeZero();
});
