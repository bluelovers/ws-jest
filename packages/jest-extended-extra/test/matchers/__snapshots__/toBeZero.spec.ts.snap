// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throw 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).not.toBeZero()[22m

Expected value to not be a zero received:
  [31m-0[39m"
`;
